graph/middleware/knowledge.py

```
@@ -145,6 +164,123 @@ def _count_tokens(text: str) -> int:
+        Args:
+            query: Combined user message + recent context (up to 2K chars).
+            k:     Max results; defaults to self._skills_top_k.
+        """
+        if self._skills_index is None:
+            return []
+        if not query or not query.strip():
+            return []
+        k = k if k is not None else self._skills_top_k
+        try:
+            return self._skills_index.load_skills(query, k=k)
+        except Exception as exc:  # pragma: no cover
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)